### PR TITLE
media-video/qgifer: changing dependency from <media-libs/giflib-4.2.3:0

### PR DESCRIPTION
**Pull request by ChaosEngine**
to current version media-libs/giflib-5*
Solves Gentoo-Bug: https://bugs.gentoo.org/536634

### CODE REVIEW

File: media-video/qgifer/Manifest [hunk 1]
```diff
@@ -1 +1,7 @@
...
 DIST qgifer-0.2.1-source.tar.gz 373441 SHA256 b05f2b44faad2a94a746eb7de411875f133e6038ec81a5f3247c86a25a41e9f0 SHA512 ce48e5756cbab283d5894ddb369114e52c9cedc2fe15aed5d22aeeacc822058123d3a2f3c9a6eddf94a1eef992c3ddcad77d6941c76ba0c795c8b44209fac7df WHIRLPOOL 4095c8a6a373b234007b3015cef9ce9d4a023cf80e2701cec7ad5b2df1ab51e28cf63becb412d25e7305571e284b332542b8e15a7d49bd2d40ffe239753be5e4
+EBUILD qgifer-0.2.1-r5.ebuild 1099 SHA256 0eacb4226b595687e48c452e1e214b6b325ba7ecdafd0f554779407009d683ea SHA512 9f483d843a32c929086b2a620b07eac5c831d638054919be871450bbe951c3a56baf89ba1aa243a2ece534b3e5d9d26fda19601872a7399040dd52c3c0b23b46 WHIRLPOOL 8c25864571ccf3ce60387332f7f812ebd57029d65d06c7f639ecb98f929ba1f6ef9d6772188c80924c70ed8140e14e007ff94f38a975bffb23225fed22ba3e80
+MISC metadata.xml 661 SHA256 7655e8668e7a47aab71ed16ffc79456112bba988acc2f67e05b8b0490feeb5d5 SHA512 71689d9ad47bfcd52ce9334dc84cbcd8696cc1896d16a13ecd5f45a2d76fff68a59791c6d08b623549ad28610daed000dcd4ebd7f8d5023cc139753e6614f8ff WHIRLPOOL 953398c12acef5172d5305ed50f574cf6957d8f0cf519f5840297df0a627b73dcd472f49677d66d0f1d16eb39effc7632b8046dcb3a905ac9981a68583dc395c
```
Comment (SoapGentoo) [inline media-video/qgifer/Manifest]: this is a fat manifest. The gentoo tree itself uses thin manifests. Use `repoman manifest` to generate thin manifests

File: media-video/qgifer/qgifer-0.2.1-r5.ebuild [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+KEYWORDS="~amd64 ~x86"
+
+IUSE="debug imagemagick opencv3"
+
+RDEPEND="media-libs/giflib:=
+	dev-qt/qtcore:4
+	dev-qt/qtgui:4
+	imagemagick? ( media-gfx/imagemagick:0 )
+	!opencv3? ( <media-libs/opencv-3.0.0:0[ffmpeg] )
+	opencv3? ( >=media-libs/opencv-3.0.0:0[ffmpeg] )
```
Comment (SoapGentoo) [inline media-video/qgifer/qgifer-0.2.1-r5.ebuild]: opencv requires `:0=`

**Comment by ChaosEngine** [comment]
@SoapGentoo made those changes, sorry for a mess.

**Comment by SoapGentoo** [comment]
@ChaosEngine last question: that patch changes the interface, i.e., in the current state it won't compile against giflib-4 anymore, right? If so, you need to `>= -5.0` depend on giflib.

**Comment by ChaosEngine** [comment]
Do we need to care for giflib-4, now that there is stable giflib-5*? If so, yes I will make it work for all the versions of giflib.

**Comment by mgorny** [comment]
@ChaosEngine, it would be best if you could get an upstreamable patch that works against both versions. Unless upstream already dropped support for 4, then it's fine with dep change.

**Comment by ChaosEngine** [comment]
@mgorny AFAIK the project stale to say the least. Code still depends hard on giflib-4. So as world moves on to giflib-5, I wanted to support it more than giflib-4* (as @SoapGentoo suggested). I will try to support both version, though.

**Comment by ChaosEngine** [comment]
I've done the changes with introducing new USE='giflib5' which will select either giflib-4*  or giflib-5.1*.

**Comment by mgorny** [comment]
> I've done the changes with introducing new USE='giflib5' which will select either giflib-4* or giflib-5.1*.

Err, that's not the correct way of doing that. You should just detect the giflib API at build time.

However, since the package is dead upstream as you mentioned, there's really no point in supporting old version. Feel free to proceed whichever way you choose — either remove that support, or make it detect giflib version at build time. However, I heard @SoapGentoo has volunteered to strip it off for you ;-D.

(nice tip: `git reflog` probably has older version of your commits)

**Comment by SoapGentoo** [comment]
@ChaosEngine thanks for the good work, keep it coming! Given the confusion about the different RDEPENDs, I removed the flag again. I've also merged the two patches into one and did some minor cosmetic edits in the ebuild. Your COMMITMSG was too long, so I cut it down.

**Comment by ChaosEngine** [comment]
Ok, thank you for your time and commitment.

**Pull request closed without merging**